src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -223,13 +225,15 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
         }
 
         /** Avoids excess round-tripping when reading status file. */
-        static class StatusCheck extends MasterToSlaveFileCallable<Integer> {
+        /* replaced static class STATUS_CHECK with anonymous to capture charset */
+        final MasterToSlaveFileCallable<Integer> STATUS_CHECK_INSTANCE = new MasterToSlaveFileCallable<Integer>() {
```

Comment:
No, you may not remote an anonymous inner class.

Reply:
Mmm... Okay, can i use the same approach as in getOutput 
```
       @Override public byte[] getOutput(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
            return getOutputFile(workspace).act(new MasterToSlaveFileCallable<byte[]>() {
                @Override public byte[] invoke(File f, VirtualChannel channel) throws IOException, InterruptedException {
```